CMakeLists.txt

```
@@ -24,82 +27,149 @@ aux_source_directory(${CMAKE_CURRENT_LIST_DIR}/src/internal ALL_SRC)
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
     PRIVATE
-    ${CMAKE_CURRENT_LIST_DIR}/lib/websocketpp
-    ${CMAKE_CURRENT_LIST_DIR}/lib/rapidjson/include
-    ${CMAKE_CURRENT_LIST_DIR}/lib/asio/asio/include
+    ${MODULE_INCLUDE_DIRS}
 )
 
-if (CMAKE_VERSION VERSION_GREATER "3.1")
-set_property(TARGET sioclient PROPERTY CXX_STANDARD 11)
-set_property(TARGET sioclient PROPERTY CXX_STANDARD_REQUIRED ON)
+if(CMAKE_VERSION VERSION_GREATER "3.1")
+    set_property(TARGET sioclient PROPERTY CXX_STANDARD 11)
+    set_property(TARGET sioclient PROPERTY CXX_STANDARD_REQUIRED ON)
 else()
-set_property(TARGET sioclient APPEND_STRING PROPERTY COMPILE_FLAGS "-std=c++11")
+    set_property(TARGET sioclient APPEND_STRING PROPERTY COMPILE_FLAGS "-std=c++11")
 endif()
+
 if(BUILD_SHARED_LIBS)
-set_target_properties(sioclient
-	PROPERTIES
-		SOVERSION ${MAJOR}
-		VERSION ${MAJOR}.${MINOR}.${PATCH}
-	)
+    set_target_properties(sioclient
+        PROPERTIES
+        SOVERSION ${MAJOR}
+        VERSION ${MAJOR}.${MINOR}.${PATCH}
+    )
 endif()
+
 list(APPEND TARGET_LIBRARIES sioclient)
 
 find_package(OpenSSL)
+
 if(OPENSSL_FOUND)
-add_library(sioclient_tls ${ALL_SRC})
-target_include_directories(sioclient_tls PUBLIC
-    ${CMAKE_CURRENT_LIST_DIR}/src 
-    PRIVATE
-    ${CMAKE_CURRENT_LIST_DIR}/lib/websocketpp
-    ${CMAKE_CURRENT_LIST_DIR}/lib/rapidjson/include
-    ${CMAKE_CURRENT_LIST_DIR}/lib/asio/asio/include
-    ${OPENSSL_INCLUDE_DIR}
-)
+    add_library(sioclient_tls ${ALL_SRC})
+    target_include_directories(sioclient_tls PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+        PRIVATE
+        ${MODULE_INCLUDE_DIRS}
+        ${OPENSSL_INCLUDE_DIR}
+    )
 
-if (CMAKE_VERSION VERSION_GREATER "3.1")
-set_property(TARGET sioclient_tls PROPERTY CXX_STANDARD 11)
-set_property(TARGET sioclient_tls PROPERTY CXX_STANDARD_REQUIRED ON)
-target_link_libraries(sioclient_tls PRIVATE ${OPENSSL_LIBRARIES} )
-else()
-set_property(TARGET sioclient_tls APPEND_STRING PROPERTY COMPILE_FLAGS "-std=c++11")
-endif()
-target_compile_definitions(sioclient_tls PRIVATE -DSIO_TLS)
-if(BUILD_SHARED_LIBS)
-set_target_properties(sioclient_tls
-	PROPERTIES
-		SOVERSION ${MAJOR}
-		VERSION ${MAJOR}.${MINOR}.${PATCH}
-	)
-endif()
-list(APPEND TARGET_LIBRARIES sioclient_tls)
+    if(CMAKE_VERSION VERSION_GREATER "3.1")
+        set_property(TARGET sioclient_tls PROPERTY CXX_STANDARD 11)
+        set_property(TARGET sioclient_tls PROPERTY CXX_STANDARD_REQUIRED ON)
+
+        if(NOT USE_SUBMODULES)
+            target_link_libraries(sioclient_tls PRIVATE OpenSSL::SSL OpenSSL::Crypto websocketpp::websocketpp asio asio::asio rapidjson)
+        else()
+            target_link_libraries(sioclient_tls PRIVATE ${OPENSSL_LIBRARIES})
+        endif()
+    else()
+        set_property(TARGET sioclient_tls APPEND_STRING PROPERTY COMPILE_FLAGS "-std=c++11")
+    endif()
```

Since we are using CMake 3.4 as minium version you can drop the else and also remove the entire if itself.

```suggestion
    set_property(TARGET sioclient_tls PROPERTY CXX_STANDARD 11)
    set_property(TARGET sioclient_tls PROPERTY CXX_STANDARD_REQUIRED ON)

    if(NOT USE_SUBMODULES)
        target_link_libraries(sioclient_tls PRIVATE OpenSSL::SSL OpenSSL::Crypto websocketpp::websocketpp asio asio::asio rapidjson)
    else()
        target_link_libraries(sioclient_tls PRIVATE ${OPENSSL_LIBRARIES})
    endif()
```